File: Data Structures & Algorithms/anagram-groups/submission-1.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();

        for (String s : strs) {
            char[] arr = new char[26];

            for (char c : s.toCharArray()) {
                arr[c - 'a']++;
            }

            String normalized = Arrays.toString(arr);

            map.computeIfAbsent(
                normalized, 
                k -> new ArrayList<String>()
            ).add(s);
        }

        return new ArrayList<>(map.values());
    }
}

File: Data Structures & Algorithms/anagram-groups/submission-1.py
from collections import defaultdict
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        m = defaultdict(list)

        for s in strs:
            fre = [0] * 26

            for c in s:
                fre[ord(c) - ord("a")] += 1
            m[tuple(fre)].append(s)

        return list(m.values())
